go/engine/deprovision_test.go

```diff
@@ -126,15 +124,27 @@ func TestDeprovision(t *testing.T) {
 	}
 }
 
-func TestDeprovisionLoggedOut(t *testing.T) {
+func TestDeprovision(t *testing.T) {
 	tc := SetupEngineTest(t, "deprovision")
 	defer tc.Cleanup()
+	if tc.G.SecretStoreAll == nil {
+		t.Fatalf("Need a secret store for this test")
+	}
+	assertDeprovisionWithSetup(t, tc)
+
+	// Now, test deprovision codepath with no secret store
+	tc.G.SecretStoreAll = nil
+	assertDeprovisionWithSetup(t, tc)
+}
+
+func assertDeprovisionLoggedOut(t *testing.T, tc libkb.TestContext) {
 
 	// Sign up a new user and have it store its secret in the
 	// secret store (if possible).
 	fu := NewFakeUserOrBust(tc.T, "dpr")
 	arg := MakeTestSignupEngineRunArg(fu)
-	arg.StoreSecret = libkb.HasSecretStore()
+
+	arg.StoreSecret = tc.G.SecretStoreAll != nil // generally true in tests
```

Comment:
this comment doesn't really make sense anymore since you're explicitly calling this function with and without SecretStoreAll...suggest just removing